articles/101_build_tool.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: default
+title: A universal build tool
+permalink: articles/build_tool.html
+abstract:
+  This article describes the rational for a universal build tool.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Preface
+
+ROS is developed with a federated model.
+It consists of many separately developed and maintained packages.
+
+The "manual" approach to build a set of packages consists of building all packages in their topological order one by one.
+For each package the documentation usually describes what the dependencies are, how to setup the environment to build the package as well as how to setup the environment afterwards to use the package.
+From an efficiency point of view that manual process does not scale well.
+
+This article describes the steps from the current build tools to a single universal build tool which automates the process and works for various different use cases.
```

Comment:
"current build tools" -> "current build tools used in the ROS ecosystem" (otherwise the scope is not clear)